tests/functional/test_freeze.py

```diff
@@ -88,11 +88,29 @@ def test_basic_freeze(script: PipTestEnvironment) -> None:
 
 
 def test_freeze_with_pip(script: PipTestEnvironment) -> None:
-    """Test pip shows itself"""
+    """Test that pip shows itself only when --all is used"""
+    result = script.pip("freeze")
+    assert "pip==" not in result.stdout
```

Comment:
This also doesn't seem directly related to this PR. It's a sensible addition, but why is it included here?

Reply:
Just making sure that pip is still hidden by default.

Reply:
A comment mentioning pip should hide itself from `freeze` would be useful

Reply:
Doesn't the test docstring say exactly that?